File: Exercise/LuoGu/Code_2023_7_23/P1161.java
package Code_2023_7_23;

import java.util.Scanner;

public class P1161 {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        int n = sc.nextInt();
        int MAX_SIZE = 2000001;
        boolean[] isOn = new boolean[MAX_SIZE];

        double a;
        int t;
        int index;

        for (int i = 0; i < n; i++) {
            a = sc.nextDouble();
            t = sc.nextInt();
            for (int j = 1; j <= t; j++) {
                index = (int) (j * a);
                isOn[index] = !isOn[index];
            }
        }

        for (int i = 1; i < MAX_SIZE; i++) {
            if (isOn[i]) {
                System.out.println(i);
                break;
            }
        }

        sc.close();

    }
}

File: Exercise/LuoGu/Code_2023_7_23/P1161.cpp
#include <iostream>
#include <algorithm>
#include <math.h>
#include <vector>
#include <string>

using namespace std;
#define MAX_SIZE 2000001
int main(void)
{

    int n;
    cin >> n;
    double a, t;
    bool is_on[MAX_SIZE];
    for (int i = 0; i < MAX_SIZE; i++)
    {
        is_on[i] = false;
    }

    int index;
    for (int i = 0; i < n; i++)
    {
        cin >> a >> t;
        for (int j = 1; j <= t; j++)
        {
            index = int(j * a);
            // cout << index << endl;
            is_on[index] = !is_on[index];
        }
        // cout << is_on[1] << endl;
    }

    for (int i = 1; i < MAX_SIZE; i++)
    {
        if (is_on[i])
        {
            cout << i << endl;
            break;
        }
    }

    return 0;
}
